fix!: correct `camelcase` rule schema for `allow` option (#18232)
eslint/eslint: docs/src/use/migrate-to-9.0.0.md

* [`no-inner-declarations` has a new default behavior with a new option](#no-inner-declarations)
* [`no-unused-vars` now defaults `caughtErrors` to `"all"`](#no-unused-vars)
* [`no-useless-computed-key` flags unnecessary computed member names in classes by default](#no-useless-computed-key)

> * [`camelcase` allow option only accept an array of string](#camelcase)

Reviewer: ```suggestion
* [`camelcase` allow option only accepts an array of strings](#camelcase)
```

**To address:** Fix the problems reported by the rule or revert to the previous behavior by setting the `enforceForClassMembers` option to `false`.

> ## <a name="camelcase"></a> `camelcase` allow option only accepts an array of strings

Reviewer: ```suggestion
**Related issue(s):** [#18042](https://github.com/eslint/eslint/issues/18042)

## <a name="camelcase"></a> `camelcase` allow option only accepts an array of strings
```

Seems that the line for a related issue for the previous item was accidentally deleted.

Author: arg, shit, fixing

Previously the camelcase rule didn't enforce the `allow` option to be an array of strings. In ESLint v9.0.0, the `allow` option now only accepts an array of strings.

> **To address:** Fix the the eventual configuration problems reported by the eslint

Reviewer: ```suggestion
**To address:** If ESLint reports invalid configuration for this rule, update your configuration.
```
